### PR TITLE
[d3d9, dxso] Refactor spec constants to use a bitfield layout 

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Part of a series of work for GPL refactoring.

### CODE REVIEW

File: src/d3d9/d3d9_spec_constants.h [hunk 1]
```diff
@@ -1,31 +1,119 @@
...
+      { 1, 23, 2 },  // PointMode
+      { 1, 25, 2 },  // VertexFogMode
+      { 1, 27, 2 },  // PixelFogMode
+      { 1, 29, 1 },  // FogEnabled
+
+      { 2, 0,  20 }, // SamplerNull
+      { 2, 20, 6 },  // ProjectionType
+
+      { 3, 0,  16 }, // VertexShaderBools
+      { 3, 16, 16 }, // VertexShaderBools
```
Comment (K0bin) [inline src/d3d9/d3d9_spec_constants.h:61]: This should probably be PixelShaderBools.

**Review by K0bin — COMMENTED** [review]
Tiny nit, looks okay to me over all.

File: src/d3d9/d3d9_spec_constants.h (hunk 1 above)
Comment (Joshua-Ashton) [reply src/d3d9/d3d9_spec_constants.h]: Fixed.

**Review by K0bin — APPROVED** [review]

File: src/d3d9/d3d9_spec_constants.h [hunk 2]
```diff
@@ -1,31 +1,119 @@
...
+
+      return (data[layout.dwordOffset] & layout.mask()) >> layout.bitOffset;
+    }
+
+    std::array<uint32_t, MaxSpecDwords> data = {};
+  };
+
+  class D3D9ShaderSpecConstantManager {
+  public:
+    uint32_t get(SpirvModule &module, D3D9SpecConstantId id) {
```
Comment (DadSchoorse) [inline src/d3d9/d3d9_spec_constants.h:92]: add an offset and range arguments to avoid the double extract for some of the bit fields?

Comment (doitsujin) [reply src/d3d9/d3d9_spec_constants.h:92]: Strongly agree.

cc636ed953f50785d42ce52b4fbbb003e91e5f36 should implement this, and also fixes a bug with depth and null samplers being broken in vertex shaders.

**Review by doitsujin — CHANGES_REQUESTED** [review]
See Schoorse's comment.

**Pull request merged**